DBSCE/Overview.md

```diff
@@ -0,0 +1,259 @@
+
+The high-level design is divided into two parts:
+
+1. Key generation and validation before the session starts (DBSC(E) is focused on this part).
+2. DBSC protocol applied with the generated keys (DBSC is focused on this part).
+
+Since we want to integrate DBSC(E) with the original design and make it as widely applicable as possible for all enterprise users, we are adding high-level design for the most possible combinations in this document. The intent is to have a specification that can be implemented by any browser vendor, and can be used by any IdP, and any Local Key Helper. As we cover different use cases DBSC(E) can be applied for, we differentiate between private and public local key helpers, since there are implications to the protocol based on the type of local key helper. For example, we expect well establised IdPs like Microsoft, Okta, Github to ship their own private local key helpers. DBSC(E) protocol also provides multiple extension points that simplify and improve performance for specific scenarios. The [DBSC(E) use cases](#dbsce-use-cases) section expands on some of these scenarios.
+
+DBSC(E) (in contrast with DBSC):
+
+![DBSC(E) Highlevel Design](<./DBSC(E).svg>)
+
+Highlights:
+
+Note: All references to RP, IDP are equivalent to `server` in the original [DBSC design](https://githuub.com/wicg/dbsc).
+
+1. **Pre-Session initiation with special headers (steps 1-2):** When a user starts a sign-in process, or initiates a session, the webpage initiating the session sends special headers `Sec-Session-GenerateKey` and `Sec-Session-HelperIdList` to the browser in response, to indicate that the session is expected to be DBSC(E) compliant.
+
+   - The `Sec-Session-GenerateKey` header contains the URL of the server(RP), the URL of the IdP (authentication service in most cases - which is optional for consumer use cases), a `nonce`(challenge) and any extra parameters that the IdP wants to send to the Local Key Helper.
+   - As the Local Key Helper is public or third party, and potentially can be from a different vendor, `nonce`(challenge) is essential to prevent replay of cached binding key/binding statements from a different device and to prevent the clock-skew between the IdP and the Local Key Helper, especially for short lived Binding statements. `nonce`(challenge) is expected to be generated by the IdP/RP as a part of the request, a random number with time sensitivity, and is expected to be embedded in the binding statement. IdP will be able to validate nonce to ensure the binding statement is freshly issued.
```

Comment:
I think we need to provide more details why replay of the binding statement is an issue and what attack it can result in.

Reply:
Addressed